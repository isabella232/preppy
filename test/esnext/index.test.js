/* global __dirname */
import { readFile } from "fs"
import { resolve } from "path"
import pify from "pify"
import rimraf from "rimraf"

import preppy from "../../src/index"

const lazyRead = pify(readFile)
const lazyDelete = pify(rimraf)

jest.setTimeout(20000)

test("Publish Test File via Babel", async () => {
  await lazyDelete(resolve(__dirname, "./dist"))

  await preppy({
    root: __dirname
  })

  expect(
    await lazyRead(resolve(__dirname, "dist/index.cjs.js"), "utf8")
  ).toMatchSnapshot()
  expect(
    await lazyRead(resolve(__dirname, "dist/index.esm.js"), "utf8")
  ).toMatchSnapshot()
  expect(
    await lazyRead(resolve(__dirname, "dist/index.umd.js"), "utf8")
  ).toMatchSnapshot()
})
