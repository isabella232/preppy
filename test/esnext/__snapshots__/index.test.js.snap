// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publish Test File via Babel: cjs 1`] = `
"/*! esnext v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

require('core-js/modules/es.array.iterator');
require('core-js/modules/es.promise');
var _camelCase = _interopDefault(require('lodash/camelCase'));

console.log(\\"CherryPick Import Lodash:\\", _camelCase(\\"hello world\\") === \\"helloWorld\\");
console.log(\\"Package\\", \\"esnext\\");
console.log(\\"Target\\", \\"lib\\");
console.log(\\"ES2016 Enabled:\\", 2 ** 2 === 4);
new Promise((resolve, reject) => {
  resolve(\\"resolved\\");
}).then(first => {
  console.log(\\"Promise:\\", first);
});
var CONSTANT = 123;
console.log(\\"Constant:\\", CONSTANT);
var myArray = [1, 2, 3];
console.log(\\"Supports Array.includes?:\\", myArray.includes && myArray.includes(2));
var mySet = new Set(myArray);
console.log(\\"Supports Set:\\", mySet.add(4));

(function (supportsDefault = true) {
  console.log(\\"Supports default parameters:\\", supportsDefault);
})();
/* eslint-disable no-shadow */


var testVariable = \\"outer\\";
var _testVariable = \\"inner\\";
console.log(\\"X Value from inner scope:\\", _testVariable);
console.log(\\"X Value from outer scope:\\", testVariable);
var source = {
  first: 1,
  second: 2
};
var destructed = Object.assign({
  third: 3
}, source);
console.log(\\"Destructed:\\", destructed);

class MyClass {
  constructor() {
    console.log(\\"Called constructor\\");
    this.helper();
    this.instanceProperty = 3;
  }

  helper() {
    console.log(\\"Called helper\\");
  }

}

class SecondClass extends MyClass() {
  constructor() {
    super(100);
  }

}

console.log(\\"Initialized class:\\", new MyClass());

function helper() {
  return new Promise(function ($return, $error) {
    return Promise.resolve(PromiseHelper()).then(function ($await_1) {
      try {
        return $return(42);
      } catch ($boundEx) {
        return $error($boundEx);
      }
    }, $error);
  });
}

helper();
//# sourceMappingURL=index.cjs.js.map
"
`;

exports[`Publish Test File via Babel: esm 1`] = `
"/*! esnext v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
import 'core-js/modules/es.array.iterator';
import 'core-js/modules/es.promise';
import _camelCase from 'lodash/camelCase';

console.log(\\"CherryPick Import Lodash:\\", _camelCase(\\"hello world\\") === \\"helloWorld\\");
console.log(\\"Package\\", \\"esnext\\");
console.log(\\"Target\\", \\"lib\\");
console.log(\\"ES2016 Enabled:\\", 2 ** 2 === 4);
new Promise((resolve, reject) => {
  resolve(\\"resolved\\");
}).then(first => {
  console.log(\\"Promise:\\", first);
});
var CONSTANT = 123;
console.log(\\"Constant:\\", CONSTANT);
var myArray = [1, 2, 3];
console.log(\\"Supports Array.includes?:\\", myArray.includes && myArray.includes(2));
var mySet = new Set(myArray);
console.log(\\"Supports Set:\\", mySet.add(4));

(function (supportsDefault = true) {
  console.log(\\"Supports default parameters:\\", supportsDefault);
})();
/* eslint-disable no-shadow */


var testVariable = \\"outer\\";
var _testVariable = \\"inner\\";
console.log(\\"X Value from inner scope:\\", _testVariable);
console.log(\\"X Value from outer scope:\\", testVariable);
var source = {
  first: 1,
  second: 2
};
var destructed = Object.assign({
  third: 3
}, source);
console.log(\\"Destructed:\\", destructed);

class MyClass {
  constructor() {
    console.log(\\"Called constructor\\");
    this.helper();
    this.instanceProperty = 3;
  }

  helper() {
    console.log(\\"Called helper\\");
  }

}

class SecondClass extends MyClass() {
  constructor() {
    super(100);
  }

}

console.log(\\"Initialized class:\\", new MyClass());

function helper() {
  return new Promise(function ($return, $error) {
    return Promise.resolve(PromiseHelper()).then(function ($await_1) {
      try {
        return $return(42);
      } catch ($boundEx) {
        return $error($boundEx);
      }
    }, $error);
  });
}

helper();
//# sourceMappingURL=index.esm.js.map
"
`;

exports[`Publish Test File via Babel: umd 1`] = `
"/*! esnext v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('core-js/modules/es.array.iterator'), require('core-js/modules/es.promise'), require('lodash/camelCase')) :
  typeof define === 'function' && define.amd ? define(['core-js/modules/es.array.iterator', 'core-js/modules/es.promise', 'lodash/camelCase'], factory) :
  (global = global || self, factory(null, null, global._camelCase));
}(this, (function (es_array_iterator, es_promise, _camelCase) { 'use strict';

  _camelCase = _camelCase && Object.prototype.hasOwnProperty.call(_camelCase, 'default') ? _camelCase['default'] : _camelCase;

  console.log(\\"CherryPick Import Lodash:\\", _camelCase(\\"hello world\\") === \\"helloWorld\\");
  console.log(\\"Package\\", \\"esnext\\");
  console.log(\\"Target\\", \\"lib\\");
  console.log(\\"ES2016 Enabled:\\", 2 ** 2 === 4);
  new Promise((resolve, reject) => {
    resolve(\\"resolved\\");
  }).then(first => {
    console.log(\\"Promise:\\", first);
  });
  var CONSTANT = 123;
  console.log(\\"Constant:\\", CONSTANT);
  var myArray = [1, 2, 3];
  console.log(\\"Supports Array.includes?:\\", myArray.includes && myArray.includes(2));
  var mySet = new Set(myArray);
  console.log(\\"Supports Set:\\", mySet.add(4));

  (function (supportsDefault = true) {
    console.log(\\"Supports default parameters:\\", supportsDefault);
  })();
  /* eslint-disable no-shadow */


  var testVariable = \\"outer\\";
  var _testVariable = \\"inner\\";
  console.log(\\"X Value from inner scope:\\", _testVariable);
  console.log(\\"X Value from outer scope:\\", testVariable);
  var source = {
    first: 1,
    second: 2
  };
  var destructed = Object.assign({
    third: 3
  }, source);
  console.log(\\"Destructed:\\", destructed);

  class MyClass {
    constructor() {
      console.log(\\"Called constructor\\");
      this.helper();
      this.instanceProperty = 3;
    }

    helper() {
      console.log(\\"Called helper\\");
    }

  }

  class SecondClass extends MyClass() {
    constructor() {
      super(100);
    }

  }

  console.log(\\"Initialized class:\\", new MyClass());

  function helper() {
    return new Promise(function ($return, $error) {
      return Promise.resolve(PromiseHelper()).then(function ($await_1) {
        try {
          return $return(42);
        } catch ($boundEx) {
          return $error($boundEx);
        }
      }, $error);
    });
  }

  helper();

})));
//# sourceMappingURL=index.umd.js.map
"
`;
